### ISSUE TITLE
FlatList Clipping Issue on iOS with New Architecture Enabled

### ISSUE THREAD

**Issue by zcmgyu** [facebook/react-native#44683]
### Description

After enabling the New Architecture in a React Native project, a FlatList component is clipped by its container despite the overflow style being set to visible. This issue does not occur when the New Architecture is disabled, and the FlatList renders correctly. Additionally, the problem does not manifest on Android, even with the New Architecture enabled.

### Steps to reproduce

1. Clone the provided reproduction project.
2. Execute the command: `RCT_NEW_ARCH_ENABLED=1 npx pod-install`.
3. Execute the command: `yarn start`.
4. Execute the command: `yarn ios`.
5. Swipe list left to notice the clipping issue

### React Native Version

0.74.1

### Affected Platforms

Runtime - iOS

### Areas

Fabric - The New Renderer

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.5
  CPU: (11) arm64 Apple M3 Pro
  Memory: 121.23 MB / 18.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 21.6.1
    path: ~/.nvm/versions/node/v21.6.1/bin/node
  Yarn:
    version: 3.6.4
    path: ~/.nvm/versions/node/v21.6.1/bin/yarn
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v21.6.1/bin/npm
  Watchman:
    version: 2024.05.06.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.14.3
    path: /Users/hc794/.gem/ruby/3.3.0/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK:
    API Levels:
      - "28"
      - "33"
      - "34"
    Build Tools:
      - 30.0.3
      - 33.0.0
      - 34.0.0
    System Images:
      - android-34 | Google APIs ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2023.3 AI-233.14808.21.2331.11709847
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.11
    path: /Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home/bin/javac
  Ruby:
    version: 3.3.0
    path: /Users/hc794/.rubies/ruby-3.3.0/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.1
    wanted: 0.74.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Stacktrace or Logs

```text
Empty
```

### Reproducer

https://github.com/zcmgyu/rn-overflow-bug-reproducer

### Screenshots and Videos

**iOS Old Arch:** OK

https://github.com/facebook/react-native/assets/102789808/eafa80fc-f289-4c59-87ac-ec6d4bb1e8e0

**iOS New Arch:** NG

https://github.com/facebook/react-native/assets/102789808/f8bca067-81f4-411b-bb46-ccc9918abbb6

**Android New Arch:** OK

[Android New Arch.webm](https://github.com/facebook/react-native/assets/102789808/a81da55d-3d98-4e66-a5e3-2f01958e30dd)

**Comment by realsoelynn** [comment]
Fix just got merged. This fix will be part of next 0.74 patch release

**Comment by longnguyen2508** [comment]
@realsoelynn Oh, Thank you for promptly addressing the issue
